.github/workflows/publish-docker-image.yml

```diff
@@ -0,0 +1,48 @@
+env:
+  IMAGE_NAME: development-playground
+  REGISTRY_HOSTNAME: multiversx
+
+name: Build  & push Docker image
+
+on:
+  release:
+    types: [published]
+  pull_request:
+
+jobs:
+  build-docker-image:
+    strategy:
+      matrix:
+        runs-on: [ubuntu-latest]
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY_HOSTNAME }}/${{ env.IMAGE_NAME }}
+          
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log into Docker Hub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push image to Docker Hub
+        id: push
+        uses: docker/build-push-action@v6
+        with:
+          context: ./.devcontainer
+          file: ./.devcontainer/Dockerfile
+          platforms: linux/amd64
+          push: ${{ github.event_name != 'pull_request' }}
```

Comment:
Can we use `github.event_name == 'release' && github.event.action == 'published'`, or something similar?

Otherwise, if we update the triggers at some point without checking this condition, we might mistakenly publish images on other events.

That is, I think a check by "when we want to happen" would be better than a check by "when we don't want to happen".

Reply:
Done.